node/external/transactionAPI/gasUsedAndFeeProcessor.go

```diff
@@ -0,0 +1,69 @@
+package transactionAPI
+
+import (
+	"math/big"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/data/transaction"
+)
+
+type gasUsedAndFeeProcessor struct {
+	txFeeCalculator feeComputer
+}
+
+func newGasUsedAndFeeProcessor(txFeeCalculator feeComputer) *gasUsedAndFeeProcessor {
+	return &gasUsedAndFeeProcessor{
+		txFeeCalculator: txFeeCalculator,
+	}
+}
+
+func (gfp *gasUsedAndFeeProcessor) computeAndAttachGasUsedAndFee(tx *transaction.ApiTransactionResult) {
+	gasUsed := gfp.txFeeCalculator.ComputeGasLimit(tx)
+	fee := gfp.txFeeCalculator.ComputeTxFeeBasedOnGasUsed(tx, gasUsed)
+
+	tx.GasUsed = gasUsed
+	tx.Fee = fee.String()
+
+	if tx.IsRelayed {
+		tx.GasUsed = tx.GasLimit
+		tx.Fee = tx.InitiallyPaidFee
+	}
+
+	hasRefund := false
+	for _, scr := range tx.SmartContractResults {
+		if scr.IsRefund {
```

Comment:
might refactor:
```
if !scr.IsRefund {
   continue
}
....
break
}

Reply:
Right, or even better: split in a new function

Reply:
refactored.